pilot/pkg/bootstrap/server.go

```diff
@@ -923,6 +929,43 @@ func (s *Server) initRegistryEventHandlers() {
 	}
 }
 
+// telemetryHandler trigger to clean AccessLogging cache when telemetry logging configurations changed
+func (s *Server) telemetryHandler(prev config.Config, curr config.Config, event model.Event) {
```

Comment:
PushContext already has a caching layer (or really  it *is* the caching layer). However, in it we are just caching the logging config, not the fully generated envoy config.

Can we just instead build the envoy config as part of push context? That would avoid a lot of this complexity

Reply:
If we plan to build in push context -  we need both the actual config + generated config. I guess there may be cases where we need raw config also (in debug?) ?

Reply:
@howardjohn hope this's what's you wanted, PTAL.